src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticService.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [Shared]
+    [ExportLanguageService(typeof(MakeLocalFunctionStaticService), LanguageNames.CSharp)]
+    internal sealed class MakeLocalFunctionStaticService : ILanguageService
+    {
+        internal async Task<Document> CreateParameterSymbolAsync(Document document, LocalFunctionStatementSyntax localFunction, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(true);
+            var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localFunction, cancellationToken);
+            var dataFlow = semanticModel.AnalyzeDataFlow(localFunction);
+            var captures = dataFlow.CapturedInside;
+
+            var parameters = CreateParameterSymbol(captures);
+
+            //Finds all the call sites of the local function
+            var workspace = document.Project.Solution.Workspace;
+            var arrayNode = await SymbolFinder.FindReferencesAsync(localFunctionSymbol, document.Project.Solution, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            var rootOne = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var editor = new SyntaxEditor(rootOne, CSharpSyntaxGenerator.Instance);
+
+            foreach (var referenceSymbol in arrayNode)
+            {
+                foreach (var location in referenceSymbol.Locations)
+                {
+                    var root = await location.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var syntaxNode = root.FindNode(location.Location.SourceSpan); //Node for the identifier syntax
+
+                    var invocation = (syntaxNode as IdentifierNameSyntax).Parent as InvocationExpressionSyntax;
```

Comment:
what if it's not an identifier?  this assumption may crash now or in the future.  You should check for this, and add an error annotation to the tree indicating there is some sort of reference you don't know how to fixup.